File: tasks/implementations/cpp/tasks/tic_tac_toe.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

static const int WIN_LINES[8][3] = {{0,1,2},{3,4,5},{6,7,8},{0,3,6},{1,4,7},{2,5,8},{0,4,8},{2,4,6}};

static bool winner(const std::vector<int>& board, int player) {
    for (const auto& line : WIN_LINES) {
        if (board[line[0]] == player && board[line[1]] == player && board[line[2]] == player) return true;
    }
    return false;
}

void run_tic_tac_toe(const std::string& size, const std::string& fixturesRoot) {
    std::ifstream file(fixturesRoot + "/generated/tic_tac_toe/" + size + ".txt");
    std::string line;
    long long total = 0;
    while (std::getline(file, line)) {
        if (line.empty()) continue;
        std::stringstream ss(line);
        std::vector<int> moves;
        int value = 0;
        while (ss >> value) moves.push_back(value);
        std::vector<int> board(9, 0);
        int move_count = 0;
        int game_winner = 0;
        for (int i = 0; i < static_cast<int>(moves.size()); ++i) {
            int player = i % 2 == 0 ? 1 : 2;
            board[moves[i]] = player;
            move_count = i + 1;
            if (winner(board, player)) { game_winner = player; break; }
        }
        int board_score = 0;
        for (int i = 0; i < 9; ++i) board_score += (i + 1) * board[i];
        total += game_winner * 100LL + move_count * 10LL + board_score;
    }
    std::cout << total << std::endl;
}

File: tasks/implementations/go/tasks/tic_tac_toe.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "path/filepath"
    "strconv"
    "strings"
)

var tttLines = [8][3]int{{0,1,2},{3,4,5},{6,7,8},{0,3,6},{1,4,7},{2,5,8},{0,4,8},{2,4,6}}

func tttWinner(board [9]int, player int) bool {
    for _, line := range tttLines {
        if board[line[0]] == player && board[line[1]] == player && board[line[2]] == player { return true }
    }
    return false
}

func RunTicTacToe(size string, fixturesRoot string) {
    file, _ := os.Open(filepath.Join(fixturesRoot, "generated", "tic_tac_toe", size+".txt"))
    defer file.Close()
    scanner := bufio.NewScanner(file)
    total := 0
    for scanner.Scan() {
        line := strings.TrimSpace(scanner.Text())
        if line == "" { continue }
        parts := strings.Fields(line)
        board := [9]int{}
        moveCount := 0
        gameWinner := 0
        for idx, token := range parts {
            pos, _ := strconv.Atoi(token)
            player := 1
            if idx%2 == 1 { player = 2 }
            board[pos] = player
            moveCount = idx + 1
            if tttWinner(board, player) { gameWinner = player; break }
        }
        boardScore := 0
        for idx, value := range board { boardScore += (idx + 1) * value }
        total += gameWinner*100 + moveCount*10 + boardScore
    }
    fmt.Println(total)
}
